### ISSUE TITLE
no-param-reassign false positive

### ISSUE THREAD

**Issue by fxlemire** [progre/tslint-config-airbnb#23]
```
const myObject: {[myKey: string]: number} = {};

let myKey = '3';

if (myKey === '3') {
  myKey = '1'; // false positive. not flagged if first line is commented out
}

const myFunction = (myParam) => {
  myParam = 2; // should be the only one flagged
};
```

Possible fix: change `"no-param-reassign": true` to `"no-parameter-reassignment": true` in tslint.js

Temporary fix: in your tslint.json:
```
"no-parameter-reassignment": true,
"no-param-reassign": false // disable from tslint-config-airbnb
```

Versions:
```
"tslint": "^5.8.0",
"tslint-config-airbnb": "^5.3.0",
"tslint-eslint-rules": "^4.1.1",
"typescript": "^2.6.1"
```

**Comment by progre** [comment]
Thx for your report.
I found the cause from;
https://github.com/vrsource/vrsource-tslint-rules/commit/e025e7bb74723740f80b75863869ee96510b1596

This problem will be fixed soon.

**Comment by progre** [comment]
fixed: https://github.com/progre/tslint-config-airbnb/commit/c91dcffb6f7789a2060ecce818722e22fdbcfa12
